src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -199,14 +199,45 @@ public void ApplyLanguageRefinement(GenerationConfiguration config, CodeNamespac
 
         public async Task CreateLanguageSourceFilesAsync(GenerationLanguage language, CodeNamespace generatedCode)
         {
-            var languageWriter = LanguageWriter.GetLanguageWriter(language, this.config.OutputPath, this.config.ClientNamespaceName);
+            var languageWriter = LanguageWriter.GetLanguageWriter(language, this.config.OutputPath, this.config.ClientNamespaceName, this.config.UsesBackingStore);
             var stopwatch = new Stopwatch();
             stopwatch.Start();
             await CodeRenderer.RenderCodeNamespaceToFilePerClassAsync(languageWriter, generatedCode);
             stopwatch.Stop();
             logger.LogTrace("{timestamp}ms: Files written to {path}", stopwatch.ElapsedMilliseconds, config.OutputPath);
         }
         private static readonly string requestBuilderSuffix = "RequestBuilder";
+        private static readonly string voidType = "void";
+        private CodeClass AddApiClientClass(CodeNamespace targetNS) {
+            var codeClass = targetNS.AddClass(new CodeClass(targetNS) { 
+                Name = config.ClientClassName,
+                ClassKind = CodeClassKind.RequestBuilder,
+                Description = "The main entry point of the SDK, exposes the configuration and the fluent API."
+            }).First();
+            var constructor = codeClass.AddMethod(new CodeMethod(codeClass) {
+                IsAsync = false,
+                Name = "constructor",
+                Access = AccessModifier.Public,
+                Description = "Instantiates a new Api client and sets the default values.",
```

Comment:
optional: we have the `config.ClientClassName` so we can inject the actual class name here.

Reply:
(and we can get the class name when we need it from the parent class)